packages/gatsby-source-shopify/src/gatsby-node.js

```diff
@@ -63,28 +78,53 @@ export const sourceNodes = async (
 
     console.time(msg)
     await Promise.all([
-      createNodes(`articles`, ARTICLES_QUERY, ArticleNode, args, async x => {
-        if (x.comments)
-          await forEach(x.comments.edges, async edge =>
-            createNode(await CommentNode(imageArgs)(edge.node))
-          )
-      }),
-      createNodes(`blogs`, BLOGS_QUERY, BlogNode, args),
-      createNodes(`collections`, COLLECTIONS_QUERY, CollectionNode, args),
-      createNodes(`productTypes`, PRODUCT_TYPES_QUERY, ProductTypeNode, args),
-      createPageNodes(`pages`, PAGES_QUERY, PageNode, args),
-      createNodes(`products`, PRODUCTS_QUERY, ProductNode, args, async x => {
-        if (x.variants)
-          await forEach(x.variants.edges, async edge =>
-            createNode(await ProductVariantNode(imageArgs)(edge.node))
-          )
-
-        if (x.options)
-          await forEach(x.options, async option =>
-            createNode(await ProductOptionNode(imageArgs)(option))
-          )
-      }),
-      createShopPolicies(args),
+      ...(fetchCollections.includes(ARTICLE)
```

Comment:
Those inline ternaries are a bit hard to read. I wonder if you could separate it into normal if/else blocks? One could push promises to a list and then await for that list in the end.